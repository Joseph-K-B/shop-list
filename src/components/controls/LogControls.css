.logOut {
    background-color: #587B7F;
    border: 0;
    margin: 10%;
}

.logOut:hover {
    color: white;
}

.logIn {
    background-color: #587B7F;
    border: 0;
    margin: 10%;
}

.logIn:hover {
    color: white;
}