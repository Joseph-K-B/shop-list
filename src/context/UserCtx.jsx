import React, { useContext, createContext, useState } from 'react';

const UserCtx = createContext();

const UserProvider = ({children}) => {
    const [userName, setUserName] = useState('');

    return <UserCtx.Provider value={{ userName, setUserName }}>{children}</UserCtx.Provider>
};

const useUser = () => {
    const ctx = useContext(UserCtx);

    if(ctx === undefined) {
        throw new Error('useUser must be called within child components')
    };
    return ctx;
};

export {UserProvider, useUser}