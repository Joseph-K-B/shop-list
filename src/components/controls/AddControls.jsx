import React, { useState } from "react";
import { useUser } from "../../context/UserCtx";
import css from '../list/ShopList.css'

function AddControls({ onAdded }) {
  const {userName} = useUser();
  const[name, setName] = useState('')

  const handleSubmit = (e) => {
    e.preventDefault()
    setName('')
    onAdded(name) 
  }

  return(
    <>
    {userName ? (
      <form onSubmit={handleSubmit}>
        <input 
          type='text'
          value={name}
          aria-label='new item input'
          onChange={(e) => setName(e.target.value)}
        />
          <button 
            type="submit"
            className={css.addBtn} 
            aria-label='submit new item'
          >
            Add Item
          </button>
        </form>
        ) : ( 
        <p>Start shopping</p>
    )}
    </>
  );
};

export default AddControls;