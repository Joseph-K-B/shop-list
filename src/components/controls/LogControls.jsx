import React, { useState } from 'react';
import { useUser } from '../../context/UserCtx';

const LogControls = () => {
    const [name, setName] = useState('');
    const {userName, setUserName} = useUser();

    const newUser = () => {
        setUserName(name);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        newUser();
    };

    const nameSubmit = (
        <div>
            <input 
                type='text'
                placeholder='name'
                aria-label='name-input'
                value={name}
                onChange={(e) => setName(e.target.value)}
            />
            <button type='submit' aria-label='log-button'>Log</button>
        </div>
    );
   

    const loggedMsg = userName ? `List for ${userName}` : 'Sign in to create list';

    return (
        <>
            <section>
                <h3>{loggedMsg}</h3>
                <form onSubmit={handleSubmit}>
                    {userName ? null : nameSubmit}                    
                    {userName && (
                        <button
                            onClick={() => {
                                setName('')
                                setUserName('')
                            }}
                        >
                            Not {userName}?
                        </button>
                    )}
                </form>
            </section>
        </>
    )
}

export { LogControls }