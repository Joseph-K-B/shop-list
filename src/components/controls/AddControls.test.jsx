import React from 'react';
import { screen, render } from '@testing-library/react';
import AddControls from './AddControls';
import { UserProvider } from '../../context/UserCtx';
import userEvent from '@testing-library/user-event';
import { LogControls } from './LogControls';

it('renders controls to input item', async () => {
    const {container} = render(
        <UserProvider>
            <LogControls />
            <AddControls />
        </UserProvider>
    );

    const userInput = screen.getByLabelText(/name-input/i)
    const logBtn = screen.getByLabelText(/log-button/i)

    userEvent.type(userInput, 'Izzie');
    userEvent.click(logBtn);

    const nameInput = await screen.findByLabelText(/new item input/i);
    const submitBtn = await screen.findByLabelText(/submit new item/i);

    expect(nameInput).toBeInTheDocument();
    expect(submitBtn).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});