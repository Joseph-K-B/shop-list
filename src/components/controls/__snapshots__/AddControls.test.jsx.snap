// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders controls to input item 1`] = `
<div>
  <form>
    <input
      aria-label="new item input"
      type="text"
      value=""
    />
    <button
      aria-label="submit new item"
      type="submit"
    >
      Add Item
    </button>
  </form>
</div>
`;
